src/language.rs

```diff
@@ -68,19 +58,16 @@ macro_rules! define_languages {
             }
 
             pub fn get_colors(&self) -> Vec<Color> {
-                let color_struct = match *self {
-                    $( Language::$name => define_colors ! ( $colors ), )*
-                    Language::Unknown => define_colors!( vec![Color::White] ),
+                let colors = match *self {
+                    $( Language::$name => define_colors ! ( $($colors)* ), )*
+                    Language::Unknown => define_colors ! ( vec![Color::White] ),
                 };
-                let mut colors = color_struct.basic_colors;
-                match color_struct.true_colors {
-                    Some(true_colors) => match env::var("COLORTERM") {
-                        Ok(val) => if val.to_lowercase() == "truecolor" { colors = true_colors; },
-                        Err(_e) => { },
-                    },
-                    None => { },
-                };
-                colors
+                let use_true_color = env::var("COLORTERM");
+                if use_true_color.is_ok() && use_true_color.unwrap().to_lowercase() == "truecolor" && colors.true_colors.is_some() {
```

Comment:
Maybe extract it into a separate function and include the "24bit" test
```
fn is_truecolor_terminal() -> bool {
    env::var("COLORTERM")
        .map(|colorterm| colorterm == "truecolor" || colorterm == "24bit")
        .unwrap_or(false)
}
```

Reply:
agree, implemented in latest attempt.